### DISCUSSION TITLE
“Vote” button?

### DISCUSSION THREAD

**Discussion by hughlilly** [community/community#6174] [Discussions]
As a relatively new GitHub user, and one even newer to Discussions, I’m confused by this, which I’m going to call the “vote” button:
<img width="94" alt="GitHub vote button on my Discussion" src="https://user-images.githubusercontent.com/12046008/136684815-b3c8a68f-4391-4151-b37b-75d97585012a.png">

That's how it appears on Discussions I have created, and this is how it appears on Discussions created by another user:
<img width="93" alt="image" src="https://user-images.githubusercontent.com/12046008/136684973-ea2d8b5f-fa7f-41b7-8638-4e367c2a1743.png">

Without a tooltip to tell me, I’m not sure what clicking this button will do in either case. I understand from having now clicked it that it adds or removes my vote—but this is not obvious from the UI, in my opinion. Perhaps in addition to a tooltip the use of the arrow graphic/character needs to be re-thought.

Furthermore, why would I ever want to remove my vote _from an issue I created_? Surely the button should be hidden from the creating user on their own Discussions? (Edit: I see the vote button appears also when _browsing_ Discussions; whatever decision is made here should also include the appearance of the button there.)

Thanks
Hugh

**Comment by Taitava** [comment]
I agree - and voted your message. 👍 

My opinion is that any textless button should have a tool tip, otherwise it's bad UI design. Yes, it might be negative to say it's bad UI design, but then again, it's very easy to fix it and make it good design: add a tool tip. 🙂 

I also agree with the problem of the arrow symbol. The arrow symbol brings to my mind moving the message up somehow, not voting. So I do agree also that the symbol could be re-thought. But the main issue is the tooltip. We cannot find a symbol that everybody understands the same way, so more important is to have a textual explanation.

**Comment by mgates3** [comment]
I agree. It seems odd that there is both an up vote button and a thumbs up button — seems redundant. But the thumbs up button does have some kind of tool tip, unlike the up vote button.

**Reply by Taitava** [reply]
The thumbs up button is just a reaction, and it only appears when someone has reacted to a message with a thumbs up emoji (via the emoji button). So the emoji reactions that are displayed as buttons, do not have any other meaning than just telling how many people have reacted that way. They do not affect in which order discussions are displayed.

**Comment by thuongtruong1009** [comment]
The up vote button allows people to vote for someone's opinions in a useful order. The more upvotes they have, the more valuable their comments will be for the discussion. You can refer to the document at [discussion](https://docs.github.com/en/discussions)

**Reply by Taitava** [reply]
I still think there's room for improving here. First, the button could give at least some hint about what it is used for. Second, it's hard to search for information in documentation when there are no specific words for a particular feature. I'm not so keen to do Google searches like _"What does up arrow button do in GitHub discussions"_ - it's just too much effort only to find out what a particular button does.

I appreciate that you gave the link to the documentation, but it goes to the docs front page and there's no mention about this particular button. If you want to say that a particular information can be found from the documentation, it's strange that the provided link does not actually go to the information directly. It gives me the **feeling** that maybe there actually is no such documentation page for this button, or it's too hard to find? That being said, I did do some digging now, so here's a documentation link: https://docs.github.com/en/discussions/collaborating-with-your-community-using-discussions/participating-in-a-discussion#upvoting-a-discussion

The documentation I found is good and describes the button well. Adding a tooltip however could still make the UI more understandable, I think. 🙂

**Comment by EPronizius** [comment]


**Reply by Taitava** [reply]
Sorry, I didn't understand. What "step" term, where? 🙂

**Reply by EPronizius** [reply]
omg, wrong thread!!!

**Comment by JackKelly** [comment]
I agree with all the points made by @hughlilly at the top of this thread! There should be a tooltip on the "upvote" button.

Also, I'm confused by the overlap between the "thumbs up" reaction and the "upvote". To be frank, I don't understand why the "upvote" button exists. Why not just count the number of "thumbs ups"?

For anyone stumbling across this thread because - like me - they searched for an explanation of the "up arrow" in GitHub discussions, here's the specific place in the GitHub docs which explains the upvote button & mechanics: https://docs.github.com/en/discussions/collaborating-with-your-community-using-discussions/participating-in-a-discussion#upvoting-a-discussion

**Reply by sinatrocious** [reply]
Please GitHub, **add tooltip to that button**.

It's very confusing. I've wasted so much time when every time stumble over meaning of this mysterious button and until today, when I finally found this post and now know what it means. And worse I may forget it again and there is no way to mouseover it, read goddamit tooltip and remind myself what is it.

**Comment by reukiodo** [comment]
Why is it not available for EVERY comment and only direct replies to the main issue? I find that incredibly annoying now that there are ⬆️ and 👍 but ⬆️ only available sometimes, where 👍 is available everywhere.

**Comment by Josephward123** [comment]
Yeah, I had the same confusion at first.

On GitHub Discussions, that button is basically an upvote toggle:

Click > add your vote
Click again > remove your vote
> Why it’s confusing
No tooltip or label
Icon isn’t very clear (not obviously a “vote”)
?: Why you can unvote your own post
It’s consistent with how voting works everywhere on GitHub
You might want to change your vote later (even on your own Discussion)
-Suggestion
Add a tooltip (“Upvote / Remove vote”)
Consider clearer icon or text label

So yeah-you’re not alone, the UX could definitely be clearer

**Comment by hridya-source-2024** [comment]
if i add a comment here  will the vote for this comment -automatically counted?
Also My other doubt is: -once we create a thread/discussion Does the vote also geneated?
